payment/payment.go

```diff
@@ -83,10 +92,15 @@ func (p *Payment) WitnessPubKeyHash() string {
 	return addr
 }
 
-// ScriptHash is a method of the Payment struct to derive a base58 p2sh address
-func (p *Payment) ScriptHash() string {
-	payload := &address.Base58{p.Network.ScriptHash, p.Hash}
-	addr := address.ToBase58(payload)
+// WitnessScriptHash is a method of the Payment struct to derive a base58 p2wsh address
+func (p *Payment) WitnessScriptHash() string {
+	//Here the Version for wpkh is always 0
+	version := byte(0x00)
```

Comment:
Instead of hardcoding the version, you can get it from the very first byte of `p.Script` since we already hardcode it in there:
```suggestion
	version := p.Script[0]
```

Also, in all these functions returning an address, I guess it would be better to add a check for being sure that `Hash`/`WitnessHash`, and also `Script` for this specific case, are not `nil` or empy byte slices. What do you think?

Reply:
@altafan this is only the case if we create a payment using factory method FromPublicKey.

Reply:
Yes, you're right 😄. Ignore this change.